lib/money/money.rb

```diff
@@ -21,7 +23,7 @@ def new(value = 0, currency = nil)
     alias_method :from_amount, :new
 
     def zero
-      new(0, NullCurrency.new)
+      new(0, NULL_CURRENCY)
```

Comment:
This optimization is cool to see, but I don't think it relates to the issue with `allocate_max_amounts`. Would of been nice if this was in a separate PR or commit at least.